docs/coding-guidelines/libraries-packaging.md

```diff
@@ -0,0 +1,101 @@
+# Packaging
+
+Libraries can be packaged in one or more of the following ways: as part of the .NETCore shared framework, part of a transport package, or as a NuGet package.
+
+## .NETCore Shared framework
+
+To add a library to  the .NETCore shared framework, that library's `AssemblyName` should be added to [NetCoreAppLibrary.props](../../src/libraries/NetCoreAppLibrary.props)
+
+The library should have both a `ref` and `src` project.  It's reference assembly will be included in the ref-pack for the Microsoft.NETCore.App shared framework, and it's implementation assembly will be included in the runtime pack.
+
+Including a library in the shared framework only includes the latest framework builds of that library (`$(NetCoreAppCurrent)`).  If a library has builds for other frameworks those will only be shipped if the library also produces a [Nuget package](#nuget-package).  If a library ships both in the shared frameork and a nuget package, it may decide to exclude it's latest `$(NetCoreAppCurrent)` build from the package.  This can be done by setting `ExcludeCurrentNetCoreAppFromPackage` to true.  Libraries should take care when doing this to ensure that whatever asset in the package that would apply to `$(NetCoreAppCurrent)` is functionally equivalent to that which it replaces from the shared framework, to avoid breaking applications which reference a newer package than the shared framework.
+
+If a library should be excluded from the ref-pack and not visible to customers targeting that shared framework, the library may be included in the `NetCoreAppLibraryNoReference` property in [NetCoreAppLibrary.props](../../src/libraries/NetCoreAppLibrary.props).
+
+Libraries included in the shared framework should ensure their closure is also included in the shared framework.  This will be validated as part of the build and errors raised if any dependencies are unsatisfied.
```

Comment:
I'm not quite sure what you mean by that. Can you please elaborate?

Reply:
I tried to clarify, LMK if it's better.